sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -677,9 +677,14 @@ static inline bool _pkt_not_for_me(kernel_pid_t *iface, ipv6_hdr_t *hdr)
     if (ipv6_addr_is_loopback(&hdr->dst)) {
         return false;
     }
-    else if (*iface == KERNEL_PID_UNDEF) {
-        *iface = gnrc_ipv6_netif_find_by_addr(NULL, &hdr->dst);
-        return (*iface == KERNEL_PID_UNDEF);
+    else if ((!ipv6_addr_is_link_local(&hdr->dst)) ||
+             (*iface == KERNEL_PID_UNDEF)) {
+        kernel_pid_t if_pid = gnrc_ipv6_netif_find_by_addr(NULL, &hdr->dst);
+        if (*iface != KERNEL_PID_UNDEF) {
```

Comment:
okay, let's think about this ... to fix the @kaspar030's issue `*iface` must not be changed at all (in that case `*iface` shouldn't be `KERNEL_PID_UNDEF`). If no interface is given (`*iface == KERNEL_PID_UNDEF`) `*iface` should be overwritten. So your original comment was correct.